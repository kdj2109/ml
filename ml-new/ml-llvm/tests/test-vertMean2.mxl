void vertMean(int[][][] x, int r, int c, int l){
    int i;
    int j;
    int k;
    int t;
    int b;
    int temp;
    int[][][] before;
    int[][][] current;
    int[][][] next;
    int[][][] origin;
    int sum;

    origin = x;
    for (i = 0; i < c-1; i = i +1){
        before = origin;
       for(j = 0; j<(i*l); j = j+1){
            before = before.+;}
        current = before;
        next = before;
        for(k = 0; k < c*l; k = k + 1){
           current =  current.+;
           next = next.+;
           next = next.+;}
        for(t = 0; t<(r-2); t = t + 1){
            for(b = 0; b < l; b = b + 1){
                sum  = $before + $current + $next;
                temp = (sum/3);
                $current = temp;
                before = before.+;
                current = current.+;
                next = next.+;
            }
             for(b = 0; b<(c*(l)-2); b = b+1){
                before = before.+;
                current = current.+;
                next = next.+;}
          }
    }
}

int main(){
     int[3][4:4] mat;
      int j;
      int i;
      int k;
      int l;
      int len;
      int[][][] pic;
      len = 4*4;
      mat = [|(2,3,4), (4,5,6), (9,7,5), (2,5,2)|(3,4,5),(9,8,7),(4,5,7),(6,3,2)|(9,5,1),(10,2,3),(6,4,2),(4,5,7)|(3,2,5),(6,7,8),(8,6,3),(4,3,2)|];
      vertMean(@@@mat, mat.rows, mat.columns, 3);
      pic = @@@mat;
      for(j = 0; j < mat.rows; j = j + 1){
          for( i = 0; i < mat.columns; i = i +1){
              for(k = 0; k < 3; k = k+1){
                  l = $pic;
                  printsl(l);
                  printssl(" ");
                  pic = pic.+; 
             }
            printssl("  ");
          }
          prints("");
        }
  }
 

