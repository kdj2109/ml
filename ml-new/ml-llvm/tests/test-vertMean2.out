2 3 4   4 5 6   9 7 5   2 5 2
4 4 3   7 5 5   6 5 4   6 3 2   
9 3 3   8 4 5   6 5 3   4 5 7 
3 2 5   6 7 8   8 6 3   4 3 2
